src/stream/src/executor/exchange/input.rs

```diff
@@ -358,3 +362,18 @@ pub(crate) fn new_input(
 
     Ok(input)
 }
+
+impl DispatcherMessageBatch {
+    fn into_messages(self) -> impl Iterator<Item = DispatcherMessage> {
+        #[auto_enums::auto_enum(Iterator)]
```

Comment:
I think we can simply adopt `Either` instead of creating a new TAIT which may confuses IDEs.